sensu/plugins/check_adaptec_raid.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python
+
+# Copyright 2016, Craig Tracey <craigtracey@gmail.com>
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+
+import argparse
+import os
+import re
+import subprocess
+import sys
+
+SUCCESS = 'success'
+CRITICAL = 'critical'
+WARNING = 'warning'
+
+RETURN_STATUS = {
+    SUCCESS: 0,
+    CRITICAL: 1,
+    WARNING: 2,
+}
+DEFAULT_FAILED_RETURN = CRITICAL
+DEFAULT_ARCCONF_PATH = '/bin/arcconf'
+
+
+def exit_with_status(status, criticality):
+    return_status = None
+    return_text = None
+    if not status == CRITICAL:
+        return_text = status
+        return_status = RETURN_STATUS[status]
+    else:
+        return_text = criticality
+        return_status = RETURN_STATUS[criticality]
+    print "Check status: %s" % return_text
+    sys.exit(return_status)
+
+
+def arcconf_exists(path):
+    return os.path.exists(path)
+
+
+def check_adaptec_status(args):
+    if not arcconf_exists(args.arcconf_path):
+        print "arcconf utility no present at %s" % args.arcconf_path
+        exit_with_status(WARNING, args.criticality)
+
+    check_commands = (
+        (['GETCONFIG', '1'],
+         r"\s*Controller Status\s*:\s*(.*)",
+         "Optimal", "Controller status not optimal"),
+        (['GETCONFIG', '1', 'LD'],
+         r"\s*Status of logical device\s*:\s*(.*)",
+         "Optimal", "Logical drive(s) not optimal"),
+        (['GETCONFIG', '1', 'PD'],
+         r"\s*S.M.A.R.T. warnings\s*:\s*(.*)",
+         "0", "Physical device SMART warnings", WARNING)
+    )
+
+    for command in check_commands:
+        status = _run_command(args, *command)
+        if status != SUCCESS:
+            exit_with_status(status, args.criticality)
+    exit_with_status(SUCCESS, args.criticality)
```

Comment:
This seems like a bad idea since it will only report the top-level failure and not share any of the underlying problem if a Logical volume or physical volume is bad. As in, won't this hide the actual failure behind a component that also reports a problem?